tests/integration/run.py

```diff
@@ -0,0 +1,984 @@
+# Copyright (c) 2024 - 2024, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+"""Integration test utility."""
+
+from __future__ import annotations
+
+import argparse
+import glob
+import logging
+import logging.config
+import os
+import shutil
+import subprocess  # nosec B404
+import sys
+import time
+from abc import abstractmethod
+from collections.abc import Callable, Mapping, Sequence
+from dataclasses import dataclass
+from typing import Generic, TypedDict, TypeVar, cast
+
+import cfgv  # type: ignore
+from ruamel.yaml import YAML
+
+T = TypeVar("T")
+
+logger = logging.getLogger(sys.argv[0])
+
+environ = dict(os.environ)
+# Disable pulling the release docker image to test the locally built image instead.
+environ["DOCKER_PULL"] = "never"
+
+
+def patch_env(patch: Mapping[str, str | None]) -> dict[str, str]:
+    """Patch env."""
+    copied_env = dict(environ)  # Make a copy.
+
+    for var, value in patch.items():
+        if value is None:
+            copied_env.pop(var, None)
+        else:
+            copied_env[var] = value
+
+    return copied_env
+
+
+def configure_logging(verbose: bool) -> None:
+    """Configure logging."""
+    logging.config.dictConfig(
+        {
+            "version": 1,
+            "disable_existing_loggers": False,
+            "formatters": {
+                "standard": {
+                    "format": "[%(levelname)s|%(module)s]: %(message)s",
+                },
+                "verbose": {
+                    "format": "[%(levelname)s|%(module)s|L%(lineno)d]: %(message)s",
+                },
+            },
+            "handlers": {
+                "stderr": {
+                    "class": "logging.StreamHandler",
+                    "level": "DEBUG" if verbose else "INFO",
+                    "formatter": "verbose" if verbose else "standard",
+                    "stream": "ext://sys.stderr",
+                },
+            },
+            "root": {
+                "level": "DEBUG",
+                "handlers": ["stderr"],
+            },
+        }
+    )
+
+
+COMPARE_SCRIPTS: dict[str, Sequence[str]] = {
+    "analysis_report": ["tests", "e2e", "compare_e2e_result.py"],
+    "policy_report": ["tests", "policy_engine", "compare_policy_reports.py"],
+    "deps_report": ["tests", "dependency_analyzer", "compare_dependencies.py"],
+    "vsa": ["tests", "vsa", "compare_vsa.py"],
+}
+
+
+def check_required_file(cwd: str) -> Callable[[str], None]:
+    """Check for a required file of a test case."""
+
+    def _check(v: str) -> None:
+        filepath = os.path.join(cwd, v)
+        if not os.path.isfile(filepath):
+            raise cfgv.ValidationError(f"File {filepath} does not exist.")
+
+    return cast(Callable[[str], None], cfgv.check_and(cfgv.check_string, _check))
+
+
+def check_env(env: object) -> None:
+    """Check for a required file of a test case."""
```

Comment:
I think this doc string needs to be updated.

Reply:
Fixed in a11d365a6de339b78d91392be708900d87b9f1d4.